x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/ForceMergeActionTests.java

```diff
@@ -79,4 +75,52 @@ public void testToSteps() {
         assertThat(thirdStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, SegmentCountStep.NAME)));
         assertThat(thirdStep.getNextStepKey(), equalTo(nextStepKey));
     }
+
+    private void assertBestCompression(ForceMergeAction instance) {
+        String phase = randomAlphaOfLength(5);
+        StepKey nextStepKey = new StepKey(randomAlphaOfLength(10), randomAlphaOfLength(10), randomAlphaOfLength(10));
+        List<Step> steps = instance.toSteps(null, phase, nextStepKey);
+        assertNotNull(steps);
+        assertEquals(5, steps.size());
+        CloseIndexStep firstStep = (CloseIndexStep) steps.get(0);
+        UpdateSettingsStep secondStep = (UpdateSettingsStep) steps.get(1);
+        OpenIndexStep thirdStep = (OpenIndexStep) steps.get(2);
+        WaitForIndexColorStep fourthStep = (WaitForIndexColorStep) steps.get(3);
+        ForceMergeStep fifthStep = (ForceMergeStep) steps.get(4);
+        assertThat(firstStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, CloseIndexStep.NAME)));
+        assertThat(firstStep.getNextStepKey(), equalTo(secondStep.getKey()));
+        assertThat(secondStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, UpdateSettingsStep.NAME)));
+        assertThat(secondStep.getSettings().get(EngineConfig.INDEX_CODEC_SETTING.getKey()), equalTo(CodecService.BEST_COMPRESSION_CODEC));
+        assertThat(secondStep.getNextStepKey(), equalTo(thirdStep.getKey()));
+        assertThat(thirdStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, OpenIndexStep.NAME)));
+        assertThat(thirdStep.getNextStepKey(), equalTo(fourthStep));
+        assertThat(fourthStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, WaitForIndexColorStep.NAME)));
+        assertThat(fourthStep.getNextStepKey(), equalTo(fifthStep));
+        assertThat(fifthStep.getKey(), equalTo(new StepKey(phase, ForceMergeAction.NAME, ForceMergeStep.NAME)));
+        assertThat(fifthStep.getNextStepKey(), equalTo(nextStepKey));
+    }
+
+    public void testMissingMaxNumSegments() throws IOException {
+        BytesReference emptyObject = BytesReference.bytes(JsonXContent.contentBuilder().startObject().endObject());
+        XContentParser parser = XContentHelper.createParser(null, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+            emptyObject, XContentType.JSON);
+        Exception e = expectThrows(IllegalArgumentException.class, () -> ForceMergeAction.parse(parser));
+        assertThat(e.getMessage(), equalTo("Required [max_num_segments, best_compression]"));
+    }
+
+    public void testInvalidNegativeSegmentNumber() {
```

Comment:
We need a test for an invalid codec setting too (one where the setting is not `best_compression`)